blocks/editable/index.js

```diff
@@ -212,6 +212,42 @@ export default class Editable extends wp.element.Component {
 		if ( keyCode === BACKSPACE ) {
 			this.onSelectionChange();
 		}
+
+		if ( keyCode === ENTER && this.props.inline && this.props.onSplit ) {
+			const endNode = this.editor.selection.getEnd();
+
+			if ( endNode.nodeName !== 'BR' ) {
+				return;
+			}
+
+			const prevNode = endNode.previousSibling;
+
+			if ( ! prevNode || prevNode.nodeName !== 'BR' ) {
+				return;
+			}
+
+			const { dom } = this.editor;
+			const rootNode = this.editor.getBody();
+			const beforeRange = dom.createRng();
+			const afterRange = dom.createRng();
+
+			dom.remove( prevNode );
+
+			beforeRange.setStart( rootNode, 0 );
+			beforeRange.setEnd( endNode.parentNode, dom.nodeIndex( endNode ) );
+
+			afterRange.setStart( endNode.parentNode, dom.nodeIndex( endNode ) + 1 );
+			afterRange.setEnd( rootNode, dom.nodeIndex( rootNode.lastChild ) + 1 );
+
+			const beforeFragment = beforeRange.extractContents();
+			const afterFragment = afterRange.extractContents();
+
+			const beforeReact = nodeListToReact( [ ...beforeFragment.childNodes ], createElement );
```

Comment:
While not necessarily a guarantee, `nodeListToReact` will already handle the `NodeList` -> `Array` coercion:

https://github.com/iseulde/dom-react/blob/2159830/index.js#L116